src/server/game/Entities/Player/Player.cpp

```diff
@@ -5038,6 +5038,33 @@ void Player::ResurrectPlayer(float restore_percent, bool applySickness)
     }
 }
 
+void Player::SendGhoulResurrectRequest(Player* target)
+{
+    target->m_ghoulResurrectGUID = GetGUID();
+
+    WorldPacket data(SMSG_RESURRECT_REQUEST, (8 + 4 + 1 + 1 + 1 + 4));
+    data << uint64(GetGUID());
+    data << uint32(0);
+    data << 0;
+    data << uint8(0);
+    data << uint8(0);
+    data << uint32(0);
+    target->GetSession()->SendPacket(&data);
+}
+
+bool Player::IsValidGhoulResurrectRequest(ObjectGuid guid)
+{
+    if (!m_ghoulResurrectGUID.IsEmpty() && m_ghoulResurrectGUID == guid)
+    {
+        this->CastSpell(this, 46619 /*SPELL_DK_RAISE_ALLY*/, true, nullptr, nullptr, m_ghoulResurrectGUID);
```

Comment:
Using ObjectGuid originalCaster to take the reference for the whisper,
makes it cast "Effect 0: Id 28 (SPELL_EFFECT_SUMMON)".

Some other way to send that guid to spellscript must be used (or try a different approach).